docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+
+    // Represents the observations of a MachineDeployment's current state.
+    // +listType=map
+    // +listMapKey=type
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // Other fields...
+    // NOTE: `Phase`, `FailureReason`, `FailureMessage` fields won't be there anymore
+}
+```
+
+| v1beta1 (current)             | v1beta2 (tentative Q1 2025)                              | v1beta1 removal (tentative Q1 2026) |
+|-------------------------------|----------------------------------------------------------|-------------------------------------|
+| `UpdatedReplicas`             | `UpToDateReplicas` (renamed)                             | `UpToDateReplicas`                  |
+| `Phase` (deprecated)          | `BackCompatibilty.Phase` (renamed) (deprecated)          | (removed)                           |
+| `FailureReason` (deprecated)  | `BackCompatibilty.FailureReason` (renamed) (deprecated)  | (removed)                           |
+| `FailureMessage` (deprecated) | `BackCompatibilty.FailureMessage` (renamed) (deprecated) | (removed)                           |
+| `Conditions`                  | `BackCompatibilty.Conditions` (renamed) (deprecated)     | (removed)                           |
+| `ExprimentalConditions` (new) | `Conditions` (renamed)                                   | `Conditions`                        |
+| other fields...               | other fields...                                          | other fields...                     |
+
+Notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes.
+
+#### MachineDeployment (New)Conditions
+
+| Condition        | Note                                                                                                                                                                                                                                                   |
+|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Available`      | True if the MachineDeployment has minimum availability according to parameters specified in the deployment strategy, e.g. If using RollingUpgrade strategy, availableReplicas must be greater or equal than desired replicas - MaxUnavailable replicas |  
+| `ReplicaFailure` | This condition surfaces issues on creating Machines controlled by this MachineDeployment, if any.                                                                                                                                                      |
```

Comment:
Does this bubble up all the failure messages as well? It'd be great if so

Reply:
Clarified, also answered to a similar comment from @enxebre, PTAL

`ReplicaFailure` is just about the creation of the Machine replica in K8s.
What you are looking for (failure messages from all the machines) will be `MachinesReady`, and it will bubble up messages for the entire lifecycle of the machine, not only for creation.

Reply:
I have dropped ReplicaFailure, this will surface in MachinesReady (or MachinesAvailable). Name TBD